### PR TITLE
feat: add sixteen colors theme

### PR DESCRIPTION
**Pull request by ggobbe**
**What issue does this PR address?** 
Current themes are not working on light terminal background.
Resolves https://github.com/serilog/serilog-sinks-console/issues/123

This PR builds on top of @stopdropandrew work in PR https://github.com/serilog/serilog-sinks-console/pull/118
As it looks like @stopdropandrew is away for a while, I took the liberty to start from his work and apply @nblumhardt's requests.

It adds a new theme `Sixteen` that is compatible with all the default unix terminal themes, mostly by avoiding using black/white colors and using the default more heavily. 

**Sixteen**:
image

**Does this PR introduce a breaking change?** 
No breaking changes

**Please check if the PR fulfills these requirements**

* [x]  The commit follows our [guidelines](https://github.com/serilog/serilog-sinks-console/blob/dev/CONTRIBUTING.md)

### CODE REVIEW

**Comment by nblumhardt** [comment]
Thanks for picking this up! This looks great; I've added some comments - apologies if they're a bit nitpicky, though I guess we live with anything we add here for a _long_ time, so worth iterating a bit :-)

Can you also please post new dark and light screenshots, ideally using both an ANSI terminal for the ANSI Sixteen, and the old Windows console for the System themes? Cheers!

**Comment by ggobbe** [comment]
Thanks for taking the time to review :) but I don't see the comments, did you submit the review?

Also, I'm prepping the screenshots today :+1:

**Comment by ggobbe** [comment]
I've removed the System Sixteen theme, it looked ok when the command line was using a white background but not ok at all with a dark background, hence its name didn't make sense, I think it'd be better to add a System "light" theme in a later PR if someone requires it.

Here are some screenshots of the ANSI Sixteen theme, both with light and dark background.

**Windows Terminal - Dark Background**
terminal-ansi-sixteen

**Windows Terminal - Light Background**
terminal-ansi-sixteen-light

**Windows Command Line**
command-ansi-sixteen

**Review by nblumhardt — COMMENTED** [review]
Oops! I did forget to submit this, sorry :-)

**Comment by nblumhardt** [comment]
The screenshots look great! Agreed that removing the system version of the theme made sense.

I've submitted my review now - sorry :-) ... Some of my comments will be outdated, please ignore where necessary. I'll try to loop back ASAP.

Thanks again!

File: README.md [hunk 1]
```diff
@@ -43,9 +43,11 @@ The following built-in themes are available:
  * `ConsoleTheme.None` - no styling
  * `SystemConsoleTheme.Literate` - styled to replicate _Serilog.Sinks.Literate_, using the `System.Console` coloring modes supported on all Windows/.NET targets; **this is the default when no theme is specified**
  * `SystemConsoleTheme.Grayscale` - a theme using only shades of gray, white, and black
- * `AnsiConsoleTheme.Literate` - an ANSI 16-color version of the "literate" theme; we expect to update this to use 256-colors for a more refined look in future
+ * `SystemConsoleTheme.Sixteen` - a version of the "literate" theme that works with light and dark backgrounds
```
[the remark this answers is not in the document]
Comment (ggobbe) [reply README.md]: I removed the system sixteen one, I think it requires a future PR to add a "light" theme for system as it won't work with dark background.

File: src/Serilog.Sinks.Console/Sinks/SystemConsole/Themes/AnsiConsoleThemes.cs [hunk 2]
```diff
@@ -80,5 +101,26 @@ static class AnsiConsoleThemes
...
+                [ConsoleThemeStyle.String] = Cyan,
+                [ConsoleThemeStyle.Number] = Magenta,
+                [ConsoleThemeStyle.Boolean] = Blue,
+                [ConsoleThemeStyle.Scalar] = Green,
+                [ConsoleThemeStyle.LevelVerbose] = Reset,
+                [ConsoleThemeStyle.LevelDebug] = Bold,
+                [ConsoleThemeStyle.LevelInformation] = BrightCyan,
+                [ConsoleThemeStyle.LevelWarning] = BrightYellow,
+                [ConsoleThemeStyle.LevelError] = BrightRed,
+                [ConsoleThemeStyle.LevelFatal] = "\x1b[38;5;0015m\x1b[48;5;0196m",
```
[the remark this answers is not in the document]
Comment (ggobbe) [reply src/Serilog.Sinks.Console/Sinks/SystemConsole/Themes/AnsiConsoleThemes.cs:123]: I made it BrithRed (same as Error level then)

File: src/Serilog.Sinks.Console/Sinks/SystemConsole/Themes/AnsiConsoleThemes.cs [hunk 3]
```diff
@@ -18,6 +18,27 @@ namespace Serilog.Sinks.SystemConsole.Themes
 {
     static class AnsiConsoleThemes
     {
+        const string Reset = "\x1b[0m";
```
[the remark this answers is not in the document]
Comment (ggobbe) [reply src/Serilog.Sinks.Console/Sinks/SystemConsole/Themes/AnsiConsoleThemes.cs:21]: Good shout :+1:

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ggobbe) [reply README.md]: system sixteen theme removed

File: src/Serilog.Sinks.Console/Sinks/SystemConsole/Themes/AnsiConsoleThemes.cs [hunk 4]
```diff
@@ -80,5 +101,26 @@ static class AnsiConsoleThemes
                 [ConsoleThemeStyle.LevelError] = "\x1b[38;5;0197m\x1b[48;5;0238m",
                 [ConsoleThemeStyle.LevelFatal] = "\x1b[38;5;0197m\x1b[48;5;0238m",
             });
+
+        public static AnsiConsoleTheme Sixteen { get; } = new AnsiConsoleTheme(
+            new Dictionary<ConsoleThemeStyle, string>
+            {
+                [ConsoleThemeStyle.Text] = Reset,
```
[the remark this answers is not in the document]
Comment (ggobbe) [reply src/Serilog.Sinks.Console/Sinks/SystemConsole/Themes/AnsiConsoleThemes.cs:108]: I've replaced all resets by Unthemed :+1:

File: src/Serilog.Sinks.Console/Sinks/SystemConsole/Themes/SystemConsoleThemes.cs [hunk 5]
```diff
@@ -81,5 +81,26 @@ static class SystemConsoleThemes
...
+                [ConsoleThemeStyle.String] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.DarkCyan },
+                [ConsoleThemeStyle.Number] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.Magenta },
+                [ConsoleThemeStyle.Boolean] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.Blue },
+                [ConsoleThemeStyle.Scalar] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.Green },
+                [ConsoleThemeStyle.LevelVerbose] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.Black },
+                [ConsoleThemeStyle.LevelDebug] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.Black },
+                [ConsoleThemeStyle.LevelInformation] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.Cyan },
+                [ConsoleThemeStyle.LevelWarning] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.Yellow },
+                [ConsoleThemeStyle.LevelError] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.Red, },
+                [ConsoleThemeStyle.LevelFatal] = new SystemConsoleThemeStyle { Foreground = ConsoleColor.White, Background = ConsoleColor.Red },
```
[the remark this answers is not in the document]
Comment (ggobbe) [reply src/Serilog.Sinks.Console/Sinks/SystemConsole/Themes/SystemConsoleThemes.cs]: system sixteen theme removed

**Comment by ggobbe** [comment]
Thanks for all the feedback, I've applied all suggestions (I think ^^). 

Here are some screenshots on both light and dark background with these change applied (you were right, the reset wasn't necessary):

image

image

**Comment by nblumhardt** [comment]
Wonderful - thank you!

**Pull request merged**